src/org/zaproxy/zap/extension/pscanrulesAlpha/StrictTransportSecurityScanner.java

```diff
@@ -87,24 +92,39 @@ private void raiseAlert(VulnType currentVT, String evidence, HttpMessage msg, in
 	@Override
 	public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
 		long start = System.currentTimeMillis();
-		if (msg.getResponseBody().length() > 0 && 
-				msg.getResponseHeader().isText() &&
-				msg.getRequestHeader().isSecure()){ //No point reporting for non-SSL resources
+		Vector<String> STSOption = msg.getResponseHeader().getHeaders("Strict-Transport-Security");

```

Comment:
- The name of the header could be extracted to a constant (and also used when checking the meta tag).
 - Names of variables should start with lower case.